custom_components/google_home/types.py

```diff
@@ -0,0 +1,84 @@
+"""Various types used in type hints."""
+from typing import Iterable, List, Optional, Set, Tuple, TypedDict
+
+from typing_extensions import Protocol
+
+from homeassistant.helpers.entity import Entity
+
+
+class AddEntitiesCallback(Protocol):
+    """Protocol type for async_setup_entry callback"""
+
+    def __call__(
+        self, new_entities: Iterable[Entity], update_before_add: bool = False
+    ) -> None:
+        ...
```

Comment:
Fancy stuff. And i believe we'll possibly be able to use that from HA stubs (weather they official ones or 3rd party), right? :)

Reply:
Yes, I'll be working on them soon.